### PR TITLE
ENH: to_sql returns rowcount

### LINKED ISSUE #23998: to_sql return inserted row count or list
Potential feature, could `df.to_sql()` return the indexes of the inserted rows? Something like [knex.js Query Builder](https://knexjs.org/#Builder-insert)

```python
inserted = df.to_sql(con=engine, name=example_table, if_exists='replace', index=False)
```
#### Problem description
This would allow for more quickly knowing how many rows have been entered from your dataframe

#### Expected Output
Something like `inserted = 2` if 2 rows have been added. Even better would be `inserted = [24, 25]` where inserted is now a list of the inserted primary keys

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #23998
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
wow looks pretty good

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2900,13 +2918,14 @@ def to_sql(
         >>> from sqlalchemy.types import Integer
         >>> df.to_sql('integers', con=engine, index=False,
         ...           dtype={"A": Integer()})
+        3
 
         >>> engine.execute("SELECT * FROM integers").fetchall()
         [(1,), (None,), (2,)]
-        """
+        """  # noqa: 501
```
Comment (jreback) [inline pandas/core/generic.py:2925]: don't you mean E501?

File: pandas/io/sql.py [hunk 2]
```diff
@@ -664,7 +665,23 @@ def to_sql(
...
+        .. versionadded:: 1.4.0
+
+    Notes
+    -----
+    The returned rows affected is the sum of the ``rowcount`` attribute of ``sqlite3.Cursor``
+    or SQLAlchemy connectable. The returned value may not reflect the exact number of written
+    rows as stipulated in the
+    `sqlite3 <https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.rowcount>`__ or
+    `SQLAlchemy <https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.BaseCursorResult.rowcount>`__
+    """  # noqa: 501
```
Comment (jreback) [inline pandas/io/sql.py:684]: i am surprised this works

File: pandas/core/generic.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/generic.py]: Thanks. Maybe this just treats it all `noqa` for all codes.

**Pull request merged**